hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/auth/HmppsAuthContextFacade.kt

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.hmpps.kotlin.auth
+
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Component
+
+@Component
+class HmppsAuthContextFacade {
+  /**
+   * This will return null if the token hasn't come from HMPPS Auth.  This is fine for application code, but tests need to
+   * then use @WithMockAuthUser rather than using a TestingAuthenticationToken or @WithMockUser annotation.
+   */
+  val authentication: AuthAwareAuthenticationToken?
+    get() = SecurityContextHolder.getContext().authentication as? AuthAwareAuthenticationToken
+
+  /**
+   * This is nullable since this can be called from an unprotected endpoint, but in the majority of cases it should
+   * be not null.  This gets the current username from the authentication, falling back to the clientId if there
+   * isn't a username passed in.
+   */
+  val currentPrincipal: String?
+    get() = authentication?.principal
+
+  val currentRoles: Collection<GrantedAuthority?>?
+    get() = authentication?.authorities
+
+  val isClientOnly: Boolean
+    get() = authentication?.isSystemClientCredentials() ?: false
+
+  val clientId: String?
+    get() = authentication?.clientId
+
+  /**
+   * We are gradually moving away from user tokens and instead using client credentials more often.  This will be NONE
+   * if client credentials are used, even if a NOMIS username is passed in.  This means that this method isn't an
+   * adequate test to see if the user is a NOMIS user. The only test at present is to look the user up in the database
```

Comment:
Looking up in the database won't be possible for most clients of the library - better to signpost to the nomis users API? (sorry can't remember which one that is!)